components/activity_and_insights/insights/hooks.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useSelector} from 'react-redux';
+
+import {getCloudSubscription, getSubscriptionProduct} from 'mattermost-redux/selectors/entities/cloud';
+import {getConfig, getLicense} from 'mattermost-redux/selectors/entities/general';
+
+import {isCloudLicense} from 'utils/license_utils';
+import {CloudProducts} from 'utils/constants';
+
+/**
+ * Returns some checks for free trial users or starter licenses
+ */
+export function useLicenseChecks(): [boolean, boolean] {
```

Comment:
Was this combined into a single hook (vs two separate hooks like `useStarterLicenseStatus` and `useFreeTrialStatus`) to avoid duplicate selector calls?

Reply:
Yeah it was